actions/channel_actions.ts

```diff
@@ -108,9 +108,7 @@ export function searchMoreChannels(term: string, showArchivedChannels: boolean):
         const {data, error} = await dispatch(ChannelActions.searchChannels(teamId, term, showArchivedChannels));
         if (data) {
             const myMembers = getMyChannelMemberships(state);
-
-            // When searching public channels, only get channels user is not a member of
-            const channels = showArchivedChannels ? data : (data as Channel[]).filter((c) => !myMembers[c.id]);
+            const channels = hideJoinedChannels ? (data as Channel[]).filter((channel) => !myMembers[channel.id]) : data;
```

Comment:
Have you removed `showArchivedChannels` due to new Design changes?

Reply:
Actually no, I am using it here. I just changed the logic and added another param (i.e. `hideJoinedChannels`)
https://github.com/mattermost/mattermost-webapp/blob/959b985e0b9ce93ae86864aa652bb06bf5dc30b2/components/more_channels/more_channels.tsx#L165